cilium/l7policy.cc

```diff
@@ -176,13 +177,28 @@ Http::FilterHeadersStatus AccessFilter::decodeHeaders(
 
 Http::FilterHeadersStatus AccessFilter::encodeHeaders(
     Http::ResponseHeaderMap& headers, bool) {
-  log_entry_.UpdateFromResponse(headers, config_->time_source_);
-  auto logType = ::cilium::EntryType::Response;
-  if (!allowed_ && headers.Status()->value() == "403") {
-    logType = ::cilium::EntryType::Denied;
-    config_->stats_.access_denied_.inc();
+  // Accepted & forwaded requests are logged by the upstream callback. Requests can remain unlogged
+  // if they are not accepted or any other error happens and upstream callback is never called.
+  // Logging the (locally generated) response is not enough as we no longer log request headers with
+  // responses.
+  if (!allowed_) {
+    // Request was not yet logged. Log it so that the headers get logged.
+    // Default logging local errors as "forwarded".
+    // The response log will contain the locally generated HTTP error code.
+    auto logType = ::cilium::EntryType::Request;
+
+    if (headers.Status()->value() == "403") {
+      // Log as a denied request.
+      logType = ::cilium::EntryType::Denied;
```

Comment:
Should this actually be `EntryType::Denied`? Will that result in an L7 response not coming through? I only see `EntryType::Denied` in one other place: https://github.com/cilium/proxy/blob/bd81a8434f5ffac11bcda6ddb0bbd7d8d68d273c/cilium/network_filter.cc#L241 and I wonder if we should be translating L7 concepts like 403 to "Denied" at all here.

Reply:
Ah, I missed the link in the PR description to mapping the denied requests to accesslog.TypeRequest. Then this seems like it could be logical. Does this mean 403 from upstream servers will show up as denied, even though the flow actually went through?

Eg: I have an app that `curls` GitHub.com's API and gets 403. Does that mean this L7 flow will show as denied? Technically it's forwarded, and I'm just seeing 403.